projects/client/RabbitMQ.Client/src/client/api/AmqpTcpEndpoint.cs

```diff
@@ -79,7 +81,7 @@ public AmqpTcpEndpoint(string hostName, int portOrMinusOne, SslOption ssl)
         {
             HostName = hostName;
             _port = portOrMinusOne;
-            Ssl = ssl;
+            Ssl = ssl ?? new SslOption();
```

Comment:
If `null` is going to be an acceptable value for the constructor I would propose either an overload for the constructor, or make the `ssl` parameter optional to indicate to callers that null is fine.